distribution/lib/Standard/Database/0.0.0-dev/src/Internal/Postgres/Postgres_Dialect.enso

```diff
@@ -492,6 +494,66 @@ decimal_div = Base_Generator.lift_binary_op "DECIMAL_DIV" x-> y->
 decimal_mod = Base_Generator.lift_binary_op "DECIMAL_MOD" x-> y->
     x ++ " - FLOOR(CAST(" ++ x ++ " AS decimal) / CAST(" ++ y ++ " AS decimal)) * " ++ y
 
+replace : Vector Builder -> Any -> Builder
+replace args metadata =
+    input = args.at 0
+    pattern = args.at 1
+    replacement = args.at 2
+
+    input_type = metadata.at 0
+    case_sensitivity = metadata.at 1
+    only_first = metadata.at 2
+
+    expression = case input_type of
+        Text ->
+            ## To use REGEXP_REPLACE on a non-regex, we have to escape it.
+            escaped_pattern = transform_literal_param Regex.escape pattern
+            case only_first of
+                False -> case case_sensitivity of
+                    Case_Sensitivity.Insensitive _ ->
+                        Builder.code "REGEXP_REPLACE(" ++ input ++ ", " ++ escaped_pattern ++ ", " ++ replacement ++ ", 1, 0, 'i')"
+                    _ ->
+                        Builder.code "REPLACE(" ++ input ++ ", " ++ pattern ++ ", " ++ replacement ++ ")"
+                True -> case case_sensitivity of
+                    Case_Sensitivity.Insensitive _ ->
+                        Builder.code "REGEXP_REPLACE(" ++ input ++ ", " ++ escaped_pattern ++ ", " ++ replacement ++ ", 1, 1, 'i')"
+                    _ ->
+                        Builder.code "REGEXP_REPLACE(" ++ input ++ ", " ++ escaped_pattern ++ ", " ++ replacement ++ ", 1, 1)"
+        Regex ->
+            pattern_string = transform_literal_param (_.pattern_string) pattern
```

Comment:
This particular transform is applied _every time_ a Regex is passed here.

IMO passing a Regex type into the SQL IR does not seem right - the Enso Regex type is not a type recognizable by SQL. IMO we should pass a raw Text (so apply `.pattern_string` _before_ constructing the IR) and some flag indicating that we are in regex mode (this is handled by `input_type`)

Reply:
Done.